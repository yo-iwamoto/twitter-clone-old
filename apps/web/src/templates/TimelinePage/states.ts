import { atom, useRecoilValue, useSetRecoilState } from 'recoil';

export const isComposeTweetModalOpenState = atom<boolean>({
  key: 'isComposeTweetModalOpenState',
  default: false,
});

export const useIsComposeTweetModalOpenStateValue = () => {
  return useRecoilValue(isComposeTweetModalOpenState);
};

export const useSetIsComposeTweetModalOpenState = () => {
  return useSetRecoilState(isComposeTweetModalOpenState);
};
