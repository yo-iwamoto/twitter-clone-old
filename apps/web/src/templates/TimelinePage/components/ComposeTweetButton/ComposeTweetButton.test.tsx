import { ComposeTweetButton } from '.';
import { RecoilObserver } from '@/__tests__/RecoilObserver';
import { isComposeTweetModalOpenState } from '@/templates/TimelinePage/states';
import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import userEvent from '@testing-library/user-event';

describe('ComposeTweetButton', () => {
  it('正しく表示されること', () => {
    // act
    const { getByRole, asFragment } = render(
      <RecoilRoot>
        <ComposeTweetButton />
      </RecoilRoot>
    );

    // assert
    expect(getByRole('button', { name: 'ツイートする' })).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('クリックでダイアログを表示すること', async () => {
    // arrange
    const user = userEvent.setup();
    const onChange = vi.fn();
    const { getByRole } = render(
      <RecoilRoot>
        <RecoilObserver node={isComposeTweetModalOpenState} onChange={onChange} />
        <ComposeTweetButton />
      </RecoilRoot>
    );

    // act
    await user.click(getByRole('button', { name: 'ツイートする' }));

    // assert
    expect(onChange).toHaveBeenLastCalledWith(true);
  });
});
