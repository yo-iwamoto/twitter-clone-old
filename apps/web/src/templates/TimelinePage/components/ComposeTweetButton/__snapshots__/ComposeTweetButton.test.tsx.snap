// Vitest Snapshot v1

exports[`ComposeTweetButton > 正しく表示されること 1`] = `
<DocumentFragment>
  <button
    aria-label="ツイートする"
    class="fixed bottom-10 right-10"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="h-20 w-20 text-blue-400 transition-colors hover:text-blue-400/90"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 2.25c-5.385 0-9.75 4.365-9.75 9.75s4.365 9.75 9.75 9.75 9.75-4.365 9.75-9.75S17.385 2.25 12 2.25zM12.75 9a.75.75 0 00-1.5 0v2.25H9a.75.75 0 000 1.5h2.25V15a.75.75 0 001.5 0v-2.25H15a.75.75 0 000-1.5h-2.25V9z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;
